### PR TITLE
Issue #12647: ensure groovy runs when pitests fail 

### PR DESCRIPTION
**Pull request by rnveach**
Issue #12647

### CODE REVIEW

**Comment by rnveach** [comment]
Output during a failure: https://github.com/checkstyle/checkstyle/actions/runs/3926061148/jobs/6711436469#step:6:62

File: .ci/pitest.sh [hunk 1]
```diff
@@ -18,12 +18,15 @@ case $1 in
   PROFILES=$(list_profiles "${0}");
 
   if [[ $(echo "$PROFILES" | grep -w -- "${1}" | cat) != "" ]]; then
-    mvn -e --no-transfer-progress -P"$1" clean test-compile org.pitest:pitest-maven:mutationCoverage
+    RESULT=0
+    mvn -e --no-transfer-progress -P"$1" clean test-compile \
+      org.pitest:pitest-maven:mutationCoverage | RESULT=1
```
Comment (rnveach) [inline .ci/pitest.sh:23]: I don't understand why this drops the maven failure.

https://github.com/checkstyle/checkstyle/actions/runs/3926061139/jobs/6711436264
> SC2030 (info): Modification of RESULT is local (to subshell caused by pipeline).
> SC2031 (info): RESULT was modified in a subshell. That change might be lost.
I also don't know what to do to please shellcheck.

Comment (Vyom-Yadav) [reply .ci/pitest.sh:23]: Why not just `set +e` before and set it back to `set -e` . If the maven command execution fails for XYZ  reason, then the pit report won't be created (most likely) and the groovy command will subsequently fail.

Comment (romani) [reply .ci/pitest.sh:23]: better to play with "set + e", we have examples in our scripts

Comment (rnveach) [reply .ci/pitest.sh:23]: @Vyom-Yadav If you wish to take on this issue, please do. I am not familar enough with bash to resolve this.

Comment (Vyom-Yadav) [reply .ci/pitest.sh:23]: Created #12686, you can close this.

**Pull request closed without merging**